x/ccv/consumer/keeper/genesis_test.go

```diff
@@ -0,0 +1,265 @@
+		malleate   func(sdk.Context, consumerkeeper.Keeper, testutil.MockedKeepers)
+		expGenesis *consumertypes.GenesisState
+	}{
+		{
+			name: "export a new chain",
+			malleate: func(ctx sdk.Context, ck consumerkeeper.Keeper, mocks testutil.MockedKeepers) {
+				// populate the states used by a new consumer chain
+				cVal, err := consumertypes.NewCCValidator(validator.Address.Bytes(), 1, pubKey)
+				require.NoError(t, err)
+				ck.SetCCValidator(ctx, cVal)
+				ck.SetProviderClientID(ctx, clientID)
+				ck.SetPendingSlashRequests(
+					ctx,
+					slashRequests,
+				)
+
+				// set the mock calls executed during the export
+				gomock.InOrder(
+					expectGetClientStateMock(ctx, mocks, "", clientID),
+					expectLatestConsensusStateMock(ctx, mocks, clientID, validator),
+				)
+			},
+
+			expGenesis: consumertypes.NewInitialGenesisState(testutil.GetClientState(""), consensusState,
+				[]abci.ValidatorUpdate{tmtypes.TM2PB.ValidatorUpdate(validator)}, slashRequests, params),
+		},
+		{
+			name: "export a chain that has an established CCV channel",
+			malleate: func(ctx sdk.Context, ck consumerkeeper.Keeper, mocks testutil.MockedKeepers) {
+				// populate the states used by a running chain
+				cVal, err := consumertypes.NewCCValidator(validator.Address.Bytes(), 1, pubKey)
+				require.NoError(t, err)
+				ck.SetCCValidator(ctx, cVal)
+				ck.SetOutstandingDowntime(ctx, sdk.ConsAddress(validator.Address.Bytes()))
+
+				// populate the required states to simulate a completed handshake
+				ck.SetProviderClientID(ctx, clientID)
+				ck.SetProviderChannel(ctx, channelID)
+				ck.SetHeightValsetUpdateID(ctx, restartHeight, matPacket.VscId)
+				ck.SetPacketMaturityTime(ctx, matPacket.VscId, matPacket.MaturityTime)
+			},
+			expGenesis: consumertypes.NewRestartGenesisState(
+				clientID,
+				channelID,
+				[]consumertypes.MaturingVSCPacket{matPacket},
+				[]abci.ValidatorUpdate{tmtypes.TM2PB.ValidatorUpdate(validator)},
+				[]types.HeightToValsetUpdateID{{Height: restartHeight, ValsetUpdateId: matPacket.VscId}},
+				[]consumertypes.OutstandingDowntime{{ValidatorConsensusAddress: sdk.ConsAddress(validator.Address.Bytes()).String()}},
+				params,
+			),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			keeperParams := testkeeper.NewInMemKeeperParams(t)
+			// Explicitly register codec with public key interface
+			keeperParams.RegisterSdkCryptoCodecInterfaces()
+			consumerKeeper, ctx, ctrl, mocks := testkeeper.GetConsumerKeeperAndCtx(t, keeperParams)
+			defer ctrl.Finish()
+			consumerKeeper.SetParams(ctx, params)
+
+			// test setup
+			tc.malleate(ctx, consumerKeeper, mocks)
+
+			// export staet to genesis
+			gotGen := consumerKeeper.ExportGenesis(ctx)
+
+			// check the obtained genesis
+			require.EqualValues(t, tc.expGenesis, gotGen)
+		})
+	}
+}
+
+func expectLatestConsensusStateMock(ctx sdk.Context, mocks testutil.MockedKeepers, clientID string, vals ...*tmtypes.Validator) *gomock.Call {
```

Comment:
I like the pattern of defining these more readable functions! Could these be added to `testutil/keeper/expectations.go`?